### ISSUE TITLE
main.cpp

### ISSUE THREAD

**Issue by EnverIdam** [EnverIdam/Enver#1]
`
#include <cstdint>
#include <chrono>
#include <iostream>
#include <thread>

#include <memory/memory.h>
#include <sdk/offsets.h>
#include <sdk/sdk.h>
#include <game/game.h>
#include <cache/cache.h>
#include <render/render.h>
#include <features/aimbot/aimbot.h>

std::int32_t main()
{
	static const char* BINARY_NAME = "RobloxPlayerBeta.exe";

	bool found = true;
	
	if (!memory->find_process_id(BINARY_NAME))
	{
		printf("unable to find\n");
		std::this_thread::sleep_for(std::chrono::seconds(5));
		return 1;
	}
	printf("found\n");
	if (!memory->attach_to_process(BINARY_NAME)) 
	{
		printf("unable to attach\n");
		std::this_thread::sleep_for(std::chrono::seconds(5));
		return 1;
	}
	printf("attached\n");
	if (!memory->find_module_address(BINARY_NAME))
	{
		printf("unable to find main module adress\n");
		std::this_thread::sleep_for(std::chrono::seconds(5));
		return 1;
	}

	printf("base : %llx\n", memory->get_module_address());

	std::uint64_t fakedatamodel{ memory->read<std::uint64_t>(memory->get_module_address() + Offsets::FakeDataModel::Pointer)};
	printf("Found Faek\n");
	game::datamodel = rbx::instance_t(memory->read<std::uint64_t>(fakedatamodel + Offsets::FakeDataModel::RealDataModel));
	printf("Found rial\n");
	game::visengine = { memory->read<std::uint64_t>(memory->get_module_address() + Offsets::VisualEngine::Pointer) };
	printf("Found visty\n");

	printf("datamodel : 0x%llx\n", game::datamodel.adress);
	printf("visengine : 0x%llx\n", game::visengine.adress);

	game::players = {game::datamodel.find_first_child_by_class("Players")};
	printf("Found pleyr\n");
	game::workspace = {game::datamodel.find_first_child_by_class("Workspace")};
	game::lighting = { game::datamodel.find_first_child_by_class("Lighting") };
	printf("Found workspeys\n");
	game::localplayer = { memory->read<rbx::player_t>(game::players.adress + Offsets::Player::LocalPlayer)};
	printf("datamodel : 0x%llx\n", game::datamodel.adress);
	printf("visengine : 0x%llx\n", game::visengine.adress);
	printf("players : 0x%llx\n", game::players.adress);
	printf("localplayer : 0x%llx\n", game::localplayer.adress);
	printf("workspace : 0x%llx\n", game::workspace.adress);
	printf("\n%s, %s\n", game::datamodel.get_name().c_str(), game::datamodel.get_class_name().c_str());

	std::thread(cache::run).detach();

	//memory->write<float>(game::character.find_first_child_by_class("Humanoid").adress + Offsets::Humanoid::WalkspeedCheck, 500);
	//memory->write<float>(game::character.find_first_child_by_class("Humanoid").adress + Offsets::Humanoid::Walkspeed, 500);

	if (!render->create_window())
	{
		printf("failed to create window\n");
		std::this_thread::sleep_for(std::chrono::seconds(5));
		return 1;
	}

	if (!render->create_device())
	{
		printf("failed to create device\n");
		std::this_thread::sleep_for(std::chrono::seconds(5));
		return 1;
	}

	if (!render->create_imgui())
	{
		printf("failed to create imgui\n");
		std::this_thread::sleep_for(std::chrono::seconds(5));
		return 1;
	}
	
	int noclipRate = 20;
	int setNoclip = 0;

	while (true)
	{
		game::character = { memory->read<rbx::model_instance_t>(game::localplayer.adress + Offsets::Player::ModelInstance) };

		//if (noclip == true && setNoclip >= noclipRate)
		//{
			//auto children = game::character.get_children();

			//noclipRate = 20;
			//if (setNoclip < noclipRate)
			//{
				//setNoclip += 1;
			//}
			//else
			//{
				//setNoclip = 0;

				//for (auto& child : children)
				//{
					//if (memory->read<std::string>(child.adress + Offsets::Instance::ClassName) == "BasePart")
					//{
						//set_can_collide(memory->read<uint64_t>(child.adress + Offsets::BasePart::Primitive), false);
					//}
				//}
			//}
		//}
		//else
		//{
			//auto children = game::character.get_children();

			//noclipRate = 35;
			//if (setNoclip < noclipRate)
			//{
			//	setNoclip += 1;
			//}
			//else if (!noclip)
			//{
				//setNoclip = 0;

				//for (auto& child : children)
				//{
					//if (memory->read<std::string>(child.adress + Offsets::Instance::ClassName) == "BasePart")
					//{
						//if (child.get_name() == "UpperTorso" || child.get_name() == "Torso" || child.get_name() == "Head")
						//{
							//set_can_collide(memory->read<uint64_t>(child.adress + Offsets::BasePart::Primitive), false);
						//}
					//}
				//}
			//}
		//}

		if (doLoopSpeed == true)
		{
			memory->write<float>(game::character.find_first_child_by_class("Humanoid").adress + Offsets::Humanoid::WalkspeedCheck, valLoopSpeed);
			memory->write<float>(game::character.find_first_child_by_class("Humanoid").adress + Offsets::Humanoid::Walkspeed, valLoopSpeed);
		}
		if (doLoopJumpPower == true)
		{
			memory->write<float>(game::character.find_first_child_by_class("Humanoid").adress + Offsets::Humanoid::JumpPower, valLoopJumpPower);
		}

		game::camera = { memory->read<rbx::instance_t>(game::localplayer.adress + Offsets::Player::ModelInstance) };

		render->start_render();

		render->render_visuals();

		if (render->running)
		{
			render->render_menu();
		}

		render->end_render();
	}
}
`
